### PR TITLE
Update imp to importlib for py3

### LINKED ISSUE #488: deprecation warning importing imp
imp is deprecated in python 3.6.8.

https://github.com/PythonCharmers/python-future/blob/7efd55d69e93e963492170869814d88b9a6d1373/src/future/standard_library/__init__.py#L65
^ this line produces deprecation warning:

```
the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
```

### PR DESCRIPTION
**Pull request by jccurtis**
`imp` was deprecated in py36. While this package is not needed in most py3 packages after Jan2020, there still are some which maintain py27 support and this `DeprecationWarning` pops up. I think I covered all cases in the src tree. There are 2 (AFAIK) locations elsewhere (examples and docs) that use `from imp import reload`. I did not patch these.

Closes #488

### CODE REVIEW

**Comment by jccurtis** [comment]
python36/37 are failing for some tests which may have to do with the `reload` behavior between `imp` and `importlib` but I haven't dug into this yet. Any thoughts?

File: tests/test_future/test_standard_library.py [hunk 1]
```diff
@@ -447,7 +447,11 @@ def test_reload(self):
         """
         reload has been moved to the imp module
         """
-        import imp
+        # imp was deprecated in python 3.6
+        if sys.version_info >= (3, 6):
+            import importlib as imp
+        else:
+            import imp
         imp.reload(imp)
```
Comment (jccurtis) [inline tests/test_future/test_standard_library.py:455]: I wonder if this is causing the CI failure.

**Comment by alanyee** [comment]
According to docs, the module has been deprecated since 3.4 with certain methods deprecated since 3.3. Looking at specific methods such as https://docs.python.org/3/library/imp.html#imp.find_module, you might not be able to do a blanket alias and have to do specific imports.

**Comment by edschofield** [comment]
Thanks @jccurtis !

**Pull request merged**